src/promptflow/promptflow/_sdk/_service/apis/collector.py

```diff
@@ -37,9 +60,23 @@ def trace_collector():
             for scope_span in resource_span.scope_spans:
                 for span in scope_span.spans:
                     # TODO: persist with batch
-                    Span._from_protobuf_object(span, resource=resource)._persist()
+                    span = Span._from_protobuf_object(span, resource=resource)
+                    span._persist()
+                    if _is_workspace_configured():
+                        _write_trace_to_cosmosdb(span, line_summary_client, span_client)
         return "Traces received", 200
 
     # JSON protobuf encoding
     elif "application/json" in content_type:
         raise NotImplementedError
+
+
+def _write_trace_to_cosmosdb(span: Span, line_summary_client, span_client):
+    from promptflow.azure._storage.cosmosdb.span import Span as SpanCosmosDB
```

Comment:
Can these work when pf[azure] is not installed but the environment variables are set?